exporter/snowflakeexporter/config.go

```diff
@@ -0,0 +1,124 @@
+// limitations under the License.
+
+package snowflakeexporter
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/snowflakedb/gosnowflake"
+	"go.opentelemetry.io/collector/config/configretry"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+)
+
+const (
+	defaultDatabase      = "otlp"
+	defaultLogsSchema    = "logs"
+	defaultMetricsSchema = "metrics"
+	defaultTracesSchema  = "traces"
+	defaultTable         = "data"
+)
+
+// Config is the config for the Snowflake exporter
+type Config struct {
+	exporterhelper.TimeoutSettings `mapstructure:",squash"`
+	exporterhelper.QueueSettings   `mapstructure:"sending_queue"`
+	configretry.BackOffConfig      `mapstructure:"retry_on_failure"`
+
+	AccountIdentifier string             `mapstructure:"account_identifier"`
+	Username          string             `mapstructure:"username"`
+	Password          string             `mapstructure:"password"`
+	Warehouse         string             `mapstructure:"warehouse"`
+	Role              string             `mapstructure:"role"`
+	Database          string             `mapstructure:"database"`
+	Parameters        map[string]*string `mapstructure:"parameters"`
+
+	Logs    TelemetryConfig `mapstructure:"logs"`
+	Metrics TelemetryConfig `mapstructure:"metrics"`
+	Traces  TelemetryConfig `mapstructure:"traces"`
+
+	dsn string
+}
+
+// TelemetryConfig is a config used by each telemetry type
+type TelemetryConfig struct {
+	Schema string `mapstructure:"schema,omitempty"`
+	Table  string `mapstructure:"table,omitempty"`
+}
+
+// Validate ensures the config is correct
+func (c *Config) Validate() error {
+	if c.AccountIdentifier == "" {
+		return errors.New("account_identifier is required")
+	}
+	if c.Username == "" {
+		return errors.New("username is required")
+	}
+	if c.Password == "" {
+		return errors.New("password is required")
+	}
+	if c.Warehouse == "" {
+		return errors.New("warehouse is required")
+	}
+	if c.Database == "" {
+		c.Database = defaultDatabase
+	}
+
+	if err := c.validateTelemetry(); err != nil {
+		return err
+	}
+
+	sf := gosnowflake.Config{}
+
+	sf.User = c.Username
+	sf.Password = c.Password
+	sf.Account = c.AccountIdentifier
+	if c.Parameters != nil {
+		sf.Params = c.Parameters
+	}
+
+	dsn, err := gosnowflake.DSN(&sf)
+	if err != nil {
+		return fmt.Errorf("failed to build DSN: %w", err)
+	}
+
+	c.dsn = dsn
+
+	return nil
+}
+
+// validateTelemetry ensures at least 1 telemetry type is configured and sets default values if needed
+func (c *Config) validateTelemetry() error {
+	if c.Logs.Schema == "" {
+		c.Logs.Schema = defaultLogsSchema
+	}
+	if c.Logs.Table == "" {
+		c.Logs.Table = defaultTable
+	}
+	if c.Metrics.Schema == "" {
+		c.Metrics.Schema = defaultMetricsSchema
+	}
+	if c.Metrics.Table == "" {
+		c.Metrics.Table = defaultTable
+	}
+	if c.Traces.Schema == "" {
+		c.Traces.Schema = defaultTracesSchema
+	}
+	if c.Traces.Table == "" {
+		c.Traces.Table = defaultTable
+	}
+
+	return nil
+}
```

Comment:
For these defaults, I think we should just set them in `factory.go` in createDefaultConfig (unless there's some issue I'm missing with that)

Reply:
I don't think there's an issue with that... the only thing I can think of is if the user intentionally sets one of those values as an empty string the collector will fail when interacting with snowflake. But we probably want to error here in that case rather than just overwriting with the default and not telling the user the config is bad